src/components/Dropdown/DropdownItem.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+
+import { UIComponent, RenderResultConfig, createShorthandFactory, customPropTypes } from '../../lib'
+import {
+  Extendable,
+  ShorthandValue,
+  ShorthandRenderFunction,
+  ComponentEventHandler,
+} from '../../../types/utils'
+import { UIComponentProps } from '../../lib/commonPropInterfaces'
+import { commonUIComponentPropTypes } from '../../lib/commonPropTypes'
+import ListItem from '../List/ListItem'
+import Image from '../Image/Image'
+
+export interface DropdownItemProps extends UIComponentProps<any, any> {
+  /** Text content for the item description. */
+  content?: string
+
+  /** The main text displayed for the item. */
+  header?: string
+
+  /** The item is highlighted inside the list. */
+  highlighted?: boolean
+
+  /** The item can have an image avatar. */
+  image?: ShorthandValue
```

Comment:
not sure that `Dropdown`'s API that has `image` prop is generic enough - I would rather expect simple Text to be a default component for the dropdown's item. In contrast, now client might think that `image` is something that is absolutely generic for dropdown's item, which is not

Reply:
Will add the first example as the one with only text items